components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -688,15 +693,45 @@ private static ByteArray generateRandom() {
         return new ByteArray(bytes);
     }
 
-    private StartRegistrationOptions buildStartRegistrationOptions(User user) {
+    private StartRegistrationOptions buildStartRegistrationOptions(User user, boolean requireResidentKey)
+            throws FIDO2AuthenticatorServerException {
 
         return StartRegistrationOptions.builder()
-                .user(buildUserIdentity(user)).build();
+                .user(buildUserIdentity(user))
+                .timeout(Integer.parseInt(userResponseTimeout))
+                .authenticatorSelection(buildAuthenticatorSelection(requireResidentKey))
+                .extensions(RegistrationExtensionInputs.builder().build())
+                .build();
+    }
+
+    private AuthenticatorSelectionCriteria buildAuthenticatorSelection(boolean requireResidentKey) {
+
+        return AuthenticatorSelectionCriteria.builder()
+                .requireResidentKey(requireResidentKey)
+                .userVerification(PREFERRED)
+                .build();
+    }
+
+    private String getUserDisplayName(User user) throws FIDO2AuthenticatorServerException {
+
+        String displayName = null;
+        String formattedNameClaimURL = "http://wso2.org/claims/formattedName";
+        try {
+            displayName = FIDO2AuthenticatorServiceComponent.getRealmService().getBootstrapRealm().getUserStoreManager()
+                    .getUserClaimValue(user.getUserName(),formattedNameClaimURL, null);
+        } catch (UserStoreException e) {
+            throw new FIDO2AuthenticatorServerException("Failed retrieving user claim",e);
```

Comment:
Improve the context of this exception by adding claim name and username info.

Reply:
Addressed with https://github.com/wso2-extensions/identity-local-auth-fido/pull/58/commits/0875f4d503afc241598249fd5c2e96a29ecc7be6